src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result: GroupsMap<T>, item: T) => {
+    const groupKey = String(item[key]); // Convert the key to a string
+
+    if (!result[groupKey]) {
+      result[groupKey] = [];
+    }
+    result[groupKey].push(item);
+
+    return result;
+  }, {});
```

Comment:
The initial value for the `reduce` method is an empty object `{}`. It's good practice to explicitly define the type of this object to match `GroupsMap<T>` for better type safety, e.g., `reduce<GroupsMap<T>>(...)`.